internal/overlord/overlord.go

```diff
@@ -112,6 +114,14 @@ func New(pebbleDir string, restartHandler restart.Handler, serviceOutput io.Writ
 	o.commandMgr = cmdstate.NewManager(o.runner)
 	o.addManager(o.commandMgr)
 
+	o.checkMgr = checkstate.NewManager()
+
+	// Tell check manager about plan updates.
+	o.serviceMgr.AddPlanHandler(o.checkMgr.Configure)
+
+	// Tell service manager about check failures.
+	o.checkMgr.AddFailureHandler(o.serviceMgr.CheckFailure)
```

Comment:
The terminology here is a bit problematic as "handlers" for us are a very specific and important idea across the entire code base: task handlers. When possible, we should try not to blur these lines to keep our conversations and minds a bit more organized.

How about:
```
o.serviceMgr.NotifyPlanChanged(o.checkMgr.PlanChanged)
o.checkMgr.NotifyCheckFailed(o.serviceMgr.CheckFailed)
```

We can try to more elegantly avoid the entanglement later on, but this looks trivial enough that we shouldn't try to over-engineer a solution just now, as there's more important feature-specific complexity in the PR for us to pay attention to.

Reply:
Good call, updated.